### PR TITLE
MOD-6150 use OS-specific installation scripts instead of readies

### PR DESCRIPTION
**Pull request by meiravgri**
* merge Python requirments for macOS and linux
* update RLTest min version to 0.7.11
* introduce build pack requirements file and install in build-artifacts flow
* task-test flow: remove install python with actions
* introduce common_installations.sh that sets up the Python environment
* mariner: remove aws cli installation

### OS-specific installation scripts:

| OS | python | gcc | additional |
|----------|----------|----------|----------|
| amzn2 | default python3 == 3.7 | 11 | openssl11
| centos7 | 3.9 | 11 | libstdc++6.0.25
| debian | default python3 == 3.9 | default == 10 |
| rocky8 | 3.11 | 11 |
| rocky9 | default python3 == 3.9 | 13 |
| ubuntu18 | 3.8 | 10 |
| ubuntu20 | default python3 == 3.8 | 10 |
| ubuntu22 | default python3 == 3.10 | default == 11 |

### CODE REVIEW

**Review by iddm — COMMENTED** [review]
Finally!

File: tests/pytests/requirements.txt [hunk 1]
```diff
@@ -3,3 +3,5 @@ gevent
 deepdiff
 conan
 RLTest >= 0.7.9
+numpy >= 1.21.6
+scipy >= 1.7.3
```
Comment (GuyAv46) [inline tests/pytests/requirements.txt:7]: are we removing the specific Linux and Mac requirements?

File: .install/ubuntu_18.04.sh [hunk 2]
```diff
@@ -6,10 +6,21 @@ MODE=$1 # whether to install using sudo or not
...
 $MODE update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 60 --slave /usr/bin/g++ g++ /usr/bin/g++-10
+
+# Install Python 3.7
+$MODE apt-get install python3.7 python3.7-venv python3-venv python3.7-dev python3-dev -y
+
+# Set python3 to point to python3.7
+$MODE update-alternatives --install /usr/bin/python python /usr/bin/python3.7 2 --slave /usr/bin/python3 python3 /usr/bin/python3.7
+
+
+#$MODE apt-get install python3-venv -y
```
Comment (GuyAv46) [inline .install/ubuntu_18.04.sh:24]: needed?

File: .install/ubuntu_18.04.sh [hunk 3]
```diff
@@ -6,10 +6,21 @@ MODE=$1 # whether to install using sudo or not
...
+
 $MODE add-apt-repository ppa:ubuntu-toolchain-r/test -y
 $MODE add-apt-repository ppa:git-core/ppa -y
 $MODE apt update
-$MODE apt install -yqq git wget make gcc-10 g++-10 openssl libssl-dev
+$MODE apt-get install -yqq build-essential git wget make gcc-10 g++-10 openssl libssl-dev
 $MODE update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 60 --slave /usr/bin/g++ g++ /usr/bin/g++-10
+
+# Install Python 3.7
+$MODE apt-get install python3.7 python3.7-venv python3-venv python3.7-dev python3-dev -y
```
Comment (GuyAv46) [inline .install/ubuntu_18.04.sh:18]: why 3.7? we already passed its end of life

File: .install/ubuntu_18.04.sh [hunk 4]
```diff
@@ -6,10 +6,21 @@ MODE=$1 # whether to install using sudo or not
...
 $MODE apt upgrade -yqq
 $MODE apt dist-upgrade -yqq
-$MODE apt install -yqq software-properties-common
+$MODE apt install -yqq software-properties-common unzip rsync
+
 $MODE add-apt-repository ppa:ubuntu-toolchain-r/test -y
 $MODE add-apt-repository ppa:git-core/ppa -y
 $MODE apt update
-$MODE apt install -yqq git wget make gcc-10 g++-10 openssl libssl-dev
+$MODE apt-get install -yqq build-essential git wget make gcc-10 g++-10 openssl libssl-dev
```
Comment (GuyAv46) [inline .install/ubuntu_18.04.sh:14]: why get?

File: .install/common_installations.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+pip install -q -r tests/pytests/requirements.txt
+# if [[ $OS_TYPE = 'Darwin' ]]
+# then
+#     pip install -q -r tests/pytests/requirements.macos.txt
+# else
+#     pip install -q -r tests/pytests/requirements.linux.txt
+# fi
+
+# These packages are needed to build the package
+# TODO: move to upload artifacts flow
```
Comment (GuyAv46) [inline .install/common_installations.sh:24]: why can't we do it now?

File: .install/common_installations.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+echo "pip version: $(pip --version)"
+echo "pip path: $(which pip)"
+
+pip install -q -r tests/pytests/requirements.txt
+# if [[ $OS_TYPE = 'Darwin' ]]
+# then
+#     pip install -q -r tests/pytests/requirements.macos.txt
+# else
+#     pip install -q -r tests/pytests/requirements.linux.txt
+# fi
```
Comment (GuyAv46) [inline .install/common_installations.sh:21]: temporary?

**Review by GuyAv46 — COMMENTED** [review]
Nice

File: .github/workflows/task-test.yml [hunk 7]
```diff
@@ -48,6 +48,8 @@ jobs:
       - name: Get Installation Mode
         id: mode
         run: |
+          echo "python --version: $(python --version)"
+          echo "python3 --version: $(python3 --version)"
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .github/workflows/task-test.yml:52]: yes and might be useful for the following PRs

File: .github/workflows/task-test.yml [hunk 8]
```diff
@@ -57,22 +59,27 @@ jobs:
...
+      - name: Setup specific
         working-directory: setup/.install
         run: ./install_script.sh ${{ steps.mode.outputs.mode }}
+
+      # - name: Setup python
+      #   id: setup-python
+      #   uses: actions/setup-python@v5
+      #   with:
+      #     python-version: "3.x"
+      - run: "echo python version: $(python3 --version)"
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .github/workflows/task-test.yml:71]: will clean up everything when all OSs script is ready

File: .install/amazon_linux_2.sh [hunk 9]
```diff
@@ -6,16 +6,16 @@ export DEBIAN_FRONTEND=noninteractive
...
 $MODE yum remove -y gcc # remove gcc 7
-$MODE yum install -y wget git openssl openssl-devel gcc10 gcc10-c++
+$MODE yum install -y wget git gcc10 gcc10-c++ python3
 $MODE update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc10-gcc 60 --slave /usr/bin/g++ g++ /usr/bin/gcc10-g++
 # Install 'openss11' and make it the default so we will use it when linking.
 # Currently it is commented out and handled on 'sbin/setup'. This is because
 # 'sbin/setup' installs 'openssl-devel' which conflicts with 'openssl11-devel'.
 # So if we install 'openssl11-devel' here we will cause 'sbin/setup' to fail.
 # Once we remove the system setup we can uncomment those lines and install 'openssl11' and 'openssl11-devel'.
 # When we do this we should also remember to remove the 'openssl' and 'openssl-devel' two lines above.
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .install/amazon_linux_2.sh:16]: not ready yet

File: .github/workflows/task-test.yml [hunk 10]
```diff
@@ -57,22 +59,27 @@ jobs:
...
-
-      - name: Setup (readies)
-        if: inputs.container != 'mcr.microsoft.com/cbl-mariner/base/core:2.0' && inputs.env != 'macos-12'
-        run: ./sbin/setup
+      - name: Setup common
+        run: |
+          echo "python3 version: $(python3 --version)"
+          echo "which python3: $(which python3)"
+          echo "which python3.6: $(which python3.6)"
+          echo "which python3.7: $(which python3.7)"
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .github/workflows/task-test.yml:81]: yes

File: tests/pytests/requirements.txt [hunk 11]
```diff
@@ -3,3 +3,5 @@ gevent
 deepdiff
 conan
 RLTest >= 0.7.9
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply tests/pytests/requirements.txt:5]: i'm aligned with master

File: tests/pytests/requirements.txt (hunk 1 above)
Comment (meiravgri) [reply tests/pytests/requirements.txt:7]: yes

File: .install/ubuntu_18.04.sh (hunk 2 above)
Comment (meiravgri) [reply .install/ubuntu_18.04.sh:24]: the comment is probably redunadnt

File: .install/ubuntu_18.04.sh (hunk 3 above)
Comment (meiravgri) [reply .install/ubuntu_18.04.sh:18]: didn't know that

File: .install/common_installations.sh (hunk 5 above)
Comment (meiravgri) [reply .install/common_installations.sh:24]: out of scope of this PR 😊

File: .install/common_installations.sh (hunk 6 above)
Comment (meiravgri) [reply .install/common_installations.sh:21]: yes

File: .github/workflows/task-test.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (meiravgri) [reply .github/workflows/task-test.yml:52]: yes
and 👍

File: .github/workflows/flow-coverage.yml [hunk 12]
```diff
@@ -19,8 +19,15 @@ jobs:
...
-      run: ./sbin/setup
+    - name: Get Installation Mode
+      id: mode
+      run: |
+        echo "mode=sudo" >> $GITHUB_OUTPUT
+    - name: Setup specific
+      working-directory: .install
+      run: ./install_script.sh ${{ steps.mode.outputs.mode }}
+    - name: Setup common
+      run: .install/common_installations.sh ${{ steps.mode.outputs.mode }}
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .github/workflows/flow-coverage.yml:30]: makes sense

File: .install/amazon_linux_2.sh [hunk 13]
```diff
@@ -4,18 +4,15 @@ set -e
...
-# Currently it is commented out and handled on 'sbin/setup'. This is because
-# 'sbin/setup' installs 'openssl-devel' which conflicts with 'openssl11-devel'.
-# So if we install 'openssl11-devel' here we will cause 'sbin/setup' to fail.
-# Once we remove the system setup we can uncomment those lines and install 'openssl11' and 'openssl11-devel'.
-# When we do this we should also remember to remove the 'openssl' and 'openssl-devel' two lines above.
-#$MODE yum install -y openssl11 openssl11-devel
-#$MODE ln -s /usr/lib64/pkgconfig/libssl11.pc /usr/lib64/pkgconfig/libssl.pc
-#$MODE ln -s /usr/lib64/pkgconfig/libcrypto11.pc /usr/lib64/pkgconfig/libcrypto.pc
-#$MODE ln -s /usr/lib64/pkgconfig/openssl11.pc /usr/lib64/pkgconfig/openssl.pc
+$MODE yum install -y http://mirror.centos.org/centos/7/extras/x86_64/Packages/centos-release-scl-rh-2-3.el7.centos.noarch.rpm
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .install/amazon_linux_2.sh:7]: a magic package that allows you to install gcc

File: .install/amazon_linux_2.sh [hunk 14]
```diff
@@ -4,18 +4,15 @@ set -e
...
+
+$MODE yum install -y wget git which devtoolset-11-gcc devtoolset-11-gcc-c++ devtoolset-11-make rsync python3 unzip
+
+source /opt/rh/devtoolset-11/enable
+
+cp /opt/rh/devtoolset-11/enable /etc/profile.d/scl-devtoolset-11.sh
+bash
+
+$MODE yum install -y openssl11 openssl11-devel
+ln -s `which openssl11` /usr/bin/openssl
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .install/amazon_linux_2.sh:17]: yes they didn't work (we tried them in JsonHDT)

File: .install/centos_linux_7.sh [hunk 15]
```diff
@@ -3,7 +3,36 @@ MODE=$1 # whether to install using sudo or not
...
+rm -f libstdc++.so.6  # remove the old symlink
+ln -s libstdc++.so.6.0.25 libstdc++.so.6
+cd -
+
+wget https://www.python.org/ftp/python/3.9.6/Python-3.9.6.tgz
+tar -xvf Python-3.9.6.tgz
+cd Python-3.9.6
+./configure --enable-optimizations
+$MODE make altinstall
+update-alternatives --install /usr/bin/python3 python3 `which python3.9` 2
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .install/centos_linux_7.sh:35]: if it's the only one then no

File: .install/common_installations.sh [hunk 16]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -e
+OS_TYPE=$(uname -s)
+MODE=$1 # whether to install using sudo or not
+
+activate_venv() {
+	echo "copy ativation script to shell config"
+	if [[ $OS_TYPE == Darwin ]]; then
+		echo "source venv/bin/activate" >> ~/.bashrc
+		echo "source venv/bin/activate" >> ~/.zshrc
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .install/common_installations.sh:10]: no, if we do that we will have to call it in every step that uses the env

File: .install/rocky_linux_9.sh [hunk 17]
```diff
@@ -3,5 +3,11 @@ MODE=$1 # whether to install using sudo or not
 set -e
 export DEBIAN_FRONTEND=noninteractive
 $MODE dnf update -y
-$MODE dnf install -y gcc gcc-c++ make wget git openssl openssl-devel
+
+$MODE dnf install -y gcc-toolset-13-gcc gcc-toolset-13-gcc-c++ make wget git openssl openssl-devel python3 which \
+    rsync unzip
+
+cp /opt/rh/gcc-toolset-13/enable /etc/profile.d/gcc-toolset-13.sh
+bash
```
[the remark this answers is not in the document]
Comment (meiravgri) [reply .install/rocky_linux_9.sh:11]: scripts in profile.d are executed upon opening a new shell (so we don't have to run the enable script on each step)
i think my intention was to run a new shell, but it's redundant here

File: .install/amazon_linux_2.sh (hunk 13 above)
[the remark this answers is not in the document]
Comment (GuyAv46) [reply .install/amazon_linux_2.sh:7]: :shushing_face: consider commenting it in the script

[the remark this answers is not in the document]
Comment (meiravgri) [reply .install/amazon_linux_2.sh:7]: it's equivalent to EPEL/ppa installation

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**